packages/protocol-kit/src/contracts/utils.ts

```diff
@@ -180,9 +189,24 @@ export async function predictSafeAddress({
     toBuffer('0x' + keccak256(toBuffer(initializer)).toString('hex') + encodedNonce)
   )
 
-  const constructorData = toBuffer(
-    ethAdapter.encodeParameters(['address'], [safeContract.getAddress()])
-  ).toString('hex')
+  const input = ethAdapter.encodeParameters(['address'], [safeContract.getAddress()])
+
+  const chainId = await ethAdapter.getChainId()
+  // zkSync Era counterfactual deployment is calculated differently
+  // https://era.zksync.io/docs/reference/architecture/differences-with-ethereum.html#create-create2
+  if ([280, 324].includes(chainId)) {
+    const bytecodeHash = zksync.utils.hashBytecode(
```

Comment:
Does the hash change over time? If not, I'd keep track of hashes for bundle size efficiency instead of saving the bytecode.

Reply:
As far as I can see it should always be the same for the same bytecode. This will add complexity at the time of adding new contract versions, as it won't just be to copy the bytecode, but converting it to the hash in an easy to store format, 3 steps.

I can document the process

Reply:
Only two more releases are planned for the Safe account contract (1.4.1 and 1.5.0 LTS), so it should be worth the effort. It's not like we're adding new proxies every week.

Reply:
And for 1.4.1/1.5.0 I think the proxy bytecode will stay the same because I don't remember changing anything in the proxy 🤔

Reply:
Added some comments to guide the next one in the process. What do you think?